projects/pastanaga/src/lib/controls/filtered-checkbox-group/filtered-checkbox-group.component.ts

```diff
@@ -150,10 +150,25 @@ export class FilteredCheckboxGroupComponent implements OnInit, OnChanges, OnDest
     }
 
     private applyFilter() {
-        this.filtered = this._checkboxes.filter(ctl => (!this.filter && !this.selectedLetter) || ctl.isFiltered);
+        this.filtered = this._checkboxes.filter(ctl => ((!this.filter && !this.selectedLetter) || ctl.isFiltered) &&
+            (this.viewAll || ctl.isSelected));
```

Comment:
this part is really hard to understand, even though I'm pretty sure it's right. could you add a comment to clarify it?

Reply:
fixed